### PR TITLE
Perform Mutesting In core/state Package

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #2910, this PR contains additional code to improve the mutation % for our `core/state` package. 

The following were gaps in our coverage:
`ExecuteStateTransition()` had extremely minimal coverage.
Most of ProcessProposerSlashings was unchecked for.
Several fields in beacon state were not checked for.
Several uncatched errors.

### CODE REVIEW

File: beacon-chain/core/state/state.go [hunk 1]
```diff
@@ -65,11 +65,16 @@ func GenesisBeaconState(deposits []*pb.Deposit, genesisTime uint64, eth1Data *pb
 		}
 	}
 
-	latestBlockRoots := make([][]byte, params.BeaconConfig().HistoricalRootsLimit)
+	latestBlockRoots := make([][]byte, params.BeaconConfig().SlotsPerHistoricalRoot)
```
Comment (prestonvanloon) [inline beacon-chain/core/state/state.go]: Why change this? It should be HistoricalRootsLimit

Comment (0xKiwi) [reply beacon-chain/core/state/state.go]: Are you sure? [Spec says SlotsPerHistoricalRoot.](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#beacon-state)

Comment (terencechain) [reply beacon-chain/core/state/state.go]: We don't have to follow every naming in the spec, in this example, we found `HistoricalRootsLimit` is more clear than `SlotsPerHistoricalRoot`

Comment (0xKiwi) [reply beacon-chain/core/state/state.go]: Ah okay, I'll leave a comment stating that. We should leave comments for any places that we go purposely off-spec.

File: beacon-chain/core/state/state.go [hunk 2]
```diff
@@ -65,12 +65,18 @@ func GenesisBeaconState(deposits []*pb.Deposit, genesisTime uint64, eth1Data *pb
 		}
 	}
 
-	latestBlockRoots := make([][]byte, params.BeaconConfig().HistoricalRootsLimit)
-	for i := 0; i < len(latestBlockRoots); i++ {
-		latestBlockRoots[i] = zeroHash
+	// Purposely set as HISTORICAL_ROOTS_LIMIT as we determined it is more clear.
```
Comment (prestonvanloon) [inline beacon-chain/core/state/state.go]: What does this comment mean?

Comment (0xKiwi) [reply beacon-chain/core/state/state.go]: I wanted to write a comment to explicitly say where we go off-spec and provide reasoning.

**Comment by prestonvanloon** [comment]
This looks good to me, except that `//beacon-chain/core/state:go_default_test ` is failing. Please take a look

**Pull request merged**